### PR TITLE
[SC-343] Disable flash loan into borrow

### PR DESCRIPTION
**Pull request by hexonaut**
Disable unused/limited use code paths to reduce attack surface.

 * Disable the flashloan into borrow feature. It is not used and introduces a lot of attack surface. The Jan 10th exploit was possible due to this code path.

### CODE REVIEW

**Comment by lucas-manuel** [comment]
Remove stable reverts

File: contracts/protocol/libraries/logic/FlashLoanLogic.sol [hunk 1]
```diff
@@ -133,40 +133,7 @@ library FlashLoanLogic {
...
-        emit FlashLoan(
-          params.receiverAddress,
-          msg.sender,
-          vars.currentAsset,
-          vars.currentAmount,
-          DataTypes.InterestRateMode(params.interestRateModes[vars.i]),
-          0,
-          params.referralCode
-        );
+        revert('FEATURE_DEPRECATED');
```
Comment (lucas-manuel) [inline contracts/protocol/libraries/logic/FlashLoanLogic.sol]: `FLASHLOAN_INTO_BORROW_DEPRECATED`

File: test-suites/pool-authorized-flashloan.spec.ts [hunk 2]
```diff
@@ -293,73 +284,6 @@ makeSuite('Pool: Authorized FlashLoan', (testEnv: TestEnv) => {
     expect(userData.currentATokenBalance).to.be.equal(expectedLiquidity, 'Invalid user balance');
   });
 
-  it('Takes out a 500 USDC flashloan with mode = 0, does not return the funds (revert expected)', async () => {
```
Comment (lucas-manuel) [inline test-suites/pool-authorized-flashloan.spec.ts:296]: Don't we need to keep this test? As far as I can tell we should only be removing mode = 1 and mode = 2

File: test-suites/pool-flashloan.spec.ts [hunk 3]
```diff
@@ -551,113 +517,6 @@ makeSuite('Pool: FlashLoan', (testEnv: TestEnv) => {
       );
   });
 
-  it('Takes out a 500 USDC flashloan with mode = 0, does not return the funds (revert expected)', async () => {
```
Comment (lucas-manuel) [inline test-suites/pool-flashloan.spec.ts:554]: Think we should keep this test as its mode 0

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: test-suites/pool-authorized-flashloan.spec.ts (hunk 2 above)
Comment (hexonaut) [reply test-suites/pool-authorized-flashloan.spec.ts:296]: Yes. The test is incorrectly labeled. It uses mode 2.

File: test-suites/pool-flashloan.spec.ts (hunk 3 above)
Comment (hexonaut) [reply test-suites/pool-flashloan.spec.ts:554]: Same here.

File: contracts/protocol/libraries/logic/FlashLoanLogic.sol (hunk 1 above)
Comment (hexonaut) [reply contracts/protocol/libraries/logic/FlashLoanLogic.sol]: changed

**Review by lucas-manuel — APPROVED** [review]

**Review by barrutko — APPROVED** [review]

**Pull request merged**